File: src/main/java/xyz/nkomarn/composter/network/protocol/packet/clientbound/ClientboundHandshakePacket.java
package xyz.nkomarn.composter.network.protocol.packet.clientbound;

import xyz.nkomarn.composter.network.WrappedBuff;

public class ClientboundHandshakePacket implements ClientboundPacket {

    private final String hash;

    public ClientboundHandshakePacket(String hash) {
        this.hash = hash;
    }

    @Override
    public void encode(WrappedBuff buff) {
        buff.writeString(hash);
    }
}

File: src/main/kotlin/kyta/composter/protocol/packet/handshaking/ClientboundHandshakePacket.kt
package kyta.composter.protocol.packet.handshaking

import kyta.composter.protocol.Packet
import kyta.composter.protocol.PacketSerializer
import kyta.composter.protocol.WriteBuffer

data class ClientboundHandshakePacket(val hash: String) : Packet {
    companion object : PacketSerializer<ClientboundHandshakePacket> {
        override fun serialize(packet: ClientboundHandshakePacket, buffer: WriteBuffer) {
            buffer.writeString(packet.hash)
        }
    }
}